airflow/ui/src/hooks/useSelectedVersion.ts

```diff
@@ -0,0 +1,85 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { useParams, useSearchParams } from "react-router-dom";
+
+import {
+  useDagRunServiceGetDagRun,
+  useDagServiceGetDagDetails,
+  useTaskInstanceServiceGetMappedTaskInstance,
+  useTaskServiceGetTask,
+} from "openapi/queries";
+import { SearchParamsKeys } from "src/constants/searchParams";
+
+const useSelectedVersion = (): string | undefined => {
+  const [searchParams] = useSearchParams();
+
+  const selectedVersionUrl = searchParams.get(SearchParamsKeys.VERSION_NUMBER);
+
+  const { dagId = "", mapIndexParam = "-1", runId = "", taskId = "" } = useParams();
```

Comment:
```suggestion
  const { dagId = "", mapIndex = "-1", runId = "", taskId = "" } = useParams();
```

The variable name needs to be an exact match to the param specified in `router.tsx`

Reply:
Nice catch thanks, I did some last minute adjustment and forgot to test again this part.